### PR TITLE
Fix STSB and WikiTexts tests

### LINKED ISSUE #1735: Caching results of the `filter` will result to inconsistent cache state
Currently it blocks, but we just got lucky:

https://github.com/pytorch/text/blob/caaa8e3c08309fe2c40ae40efaf7c2adcf1a2c8a/torchtext/datasets/stsb.py#L85

Please change the order of `filter` and `end_caching`

### PR DESCRIPTION
**Pull request by parmeet**
This PR fixes issues with STSB and WikiTexts mocked testing. In both the cases mocked files were not written properly in their respective destinations. 

fixes #1735 

cc: @Nayef211 , @VirgileHlav , @VitalyFedyunin

### CODE REVIEW

File: test/datasets/test_wikitexts.py [hunk 1]
```diff
@@ -55,19 +57,19 @@ class TestWikiTexts(TempDirMixin, TorchtextTestCase):
     root_dir = None
     samples = []
 
-    @classmethod
+    @ classmethod
```
Comment (Nayef211) [inline test/datasets/test_wikitexts.py]: nit: should we remove this space? same with the rest of the `@classmethod` and `@nested_params`  decorators

**Review by Nayef211 — APPROVED** [review]
LGTM thanks for fixing the tests!

File: test/datasets/test_wikitexts.py (hunk 1 above)
Comment (parmeet) [reply test/datasets/test_wikitexts.py]: hmm,  must be by mistake..already fixed :)

**Comment by parmeet** [comment]
> I wonder why the build is segfaulting when running `python setup.py clean`?

Ya, I am not so sure. I see same issues on audio CI (cc: @mthrok).

**Pull request merged**